src/client.rs

```diff
@@ -93,101 +67,16 @@ impl DatagoClient {
             return;
         }
 
-        // Spawn a new thread which will query the DB and send the pages
-        let pages_tx = self.pages_tx.clone();
-        let limit = self.limit;
-        let rank = self.rank;
-        let world_size = self.world_size;
-
-        assert!(
-            world_size == 0 || rank < world_size,
-            "Rank cannot be greater than or equal to world size"
-        );
-
-        let http_client = Arc::new(worker_http::new_shared_client(self.max_connections));
-
         match self.source_type {
             SourceType::Db => {
                 // convert the source_config to a SourceDBConfig
-                let source_db_config: generator_http::SourceDBConfig =
-                    serde_json::from_value(self.source_config.clone()).unwrap();
-
-                println!("Using DB as source");
-                let http_client = http_client.clone();
-
-                self.pinger = Some(thread::spawn(move || {
-                    generator_http::ping_pages(
-                        http_client,
-                        pages_tx,
-                        source_db_config,
-                        rank,
-                        world_size,
-                        limit,
-                    );
-                }));
+                self.engine = Some(generator_http::orchestrate(self));
```

Comment:
this is how that works with the PR, each specialty owns how it orchestrates, returns a DatagoEngine structure which is enough for the client to interact with (get sample or stop things)

Reply:
moved to other PR